src/widgets/scale/lv_scale_private.h

```diff
@@ -28,40 +28,49 @@ extern "C" {
  **********************/
 
 struct _lv_scale_section_t {
-    lv_style_t * main_style;
-    lv_style_t * indicator_style;
-    lv_style_t * items_style;
-    int32_t minor_range;
-    int32_t major_range;
-    uint32_t first_tick_idx_in_section;
-    uint32_t last_tick_idx_in_section;
-    uint32_t first_tick_idx_is_major;
-    uint32_t last_tick_idx_is_major;
-    int32_t first_tick_in_section_width;
-    int32_t last_tick_in_section_width;
-    lv_point_t first_tick_in_section;
-    lv_point_t last_tick_in_section;
+    lv_style_t * main_style;               /**< Style to use for MAIN part(s) of scale
+                                            *   when it falls within this section's range */
+    lv_style_t * indicator_style;          /**< Style to use for INDICATOR part(s) of scale
+                                            *   when it falls within this section's range */
+    lv_style_t * items_style;              /**< Style to use for ITEMS part(s) of scale
+                                            *   when it falls within this section's range */
+    int32_t range_min;                     /**< Scale parts with value >= this value will be drawn using applicable style. */
+    int32_t range_max;                     /**< Scale parts with value <= this value will be drawn using applicable style. */
+    uint32_t first_tick_idx_in_section;    /**< Internal (set during drawing): Tick index of first tick that falls within
+                                            *   this section; LV_SCALE_TICK_IDX_DEFAULT_ID if section contains no ticks. */
+    uint32_t last_tick_idx_in_section;     /**< Internal (set during drawing): Tick index of last tick that falls within
+                                            *   this section; LV_SCALE_TICK_IDX_DEFAULT_ID if section contains no ticks. */
+    int32_t first_tick_in_section_width;   /**< Internal (set during drawing) */
+    int32_t last_tick_in_section_width;    /**< Internal (set during drawing) */
+    lv_point_t first_tick_in_section;      /**< Internal (set during drawing) */
+    lv_point_t last_tick_in_section;       /**< Internal (set during drawing) */
+    uint32_t first_tick_idx_is_major : 1;  /**< Internal (set during drawing): true if
+                                            * `first_tick_idx_in_section` represents a major tick. */
+    uint32_t last_tick_idx_is_major  : 1;  /**< Internal (set during drawing): true if
+                                            * `last_tick_idx_in_section` represents a major tick. */
```

Comment:
We can create an a `draw_info_t` struct from these, create a local variable during drawing, and pass this local variable as a parameter to all scale draw functions. This way we would use less malloced memory for the widget and kept the the widget data cleaner too. 

E.g.
```c
lv_scale_draw_info_t draw_info = {};
...
scale_find_section_tick_idx(scale, &draw_info)
```

Reply:
> Is it okay if that becomes a separate PR?

Aboslutely!

Reply:
> > Is it okay if that becomes a separate PR?
> 
> Absolutely!

It's funny:  I dreamed about doing this in my sleep last night....  I was seeing how simple (and safe) it is to move those variables from the heap onto the stack....  😄 

I added it to my Trello tasks for LVGL so I don't forget it.  😉